src/engine_event_handler.py

```diff
@@ -24,11 +25,17 @@ def __use_next_figure(self) -> None:
     def __is_game_end(self) -> bool:
         return not self._board.is_cells_empty(self.__current_figure.points)
 
-    def handle(self, event: EngineEvent) -> GameEvent:
+    def handle(self, event: EngineEvent) -> GameEvent | None:
         if self.__is_game_end():
             return GameEvent.END
 
-        if event == EngineEvent.TIME_TICK:
+        if event == EngineEvent.PAUSE:
```

Comment:
Please move ...
```py
        if self.__is_paused:
            return
```
... into the previous `if` block and return `GameEvent.CONTINUE`

Reply:
@s373r it is DITCH